sql/core/src/test/scala/org/apache/spark/sql/CachedTableSuite.scala

```diff
@@ -837,7 +840,7 @@ class CachedTableSuite extends QueryTest with SQLTestUtils
       assert(getNumInMemoryRelations(ds) == 2)
 
       val cachedDs = sql(sqlText).cache()
-      assert(getNumInMemoryTablesRecursively(cachedDs.queryExecution.sparkPlan) == 3)
+      assert(getNumInMemoryTablesRecursively(cachedDs.queryExecution.sparkPlan) == 4)
```

Comment:
this is a interesting case, AQE can not reuse subquery accross two AQE contexts, but it seems really a corner case

Reply:
Is it fixable? `TableCacheQueryStage` can add the subqueries from the `AdaptiveSparkPlanExec` below it to the current AQE context.